src/test/java/org/seqdoop/hadoop_bam/TestVCFRoundTrip.java

```diff
@@ -0,0 +1,212 @@
+            pathname -> (!pathname.getName().startsWith(".") &&
+                !pathname.getName().startsWith("_")));
+        Arrays.sort(vcfFiles); // ensure files are sorted by name
+        for (File vcf : vcfFiles) {
+            splits++;
+            Iterators.addAll(actualVariants, parseVcf(vcf).iterator());
+            if (BGZFCodec.class.equals(codecClass)) {
+                assertTrue(BlockCompressedInputStream.isValidFile(
+                    new BufferedInputStream(new FileInputStream(vcf))));
+            } else if (BGZFEnhancedGzipCodec.class.equals(codecClass)) {
+                assertTrue(VCFFormat.isGzip(
+                    new BufferedInputStream(new FileInputStream(vcf))));
+            }
+        }
+
+        switch (expectedSplits) {
+            case EXACTLY_ONE:
+                assertEquals("Should be exactly one split", 1, splits);
+                break;
+            case MORE_THAN_ONE:
+                assertTrue("Should be more than one split", splits > 1);
+                break;
+            case ANY:
+            default:
+                break;
+        }
+
+        // use a VariantContextComparator to check variants are equal
+        VCFHeader vcfHeader = VCFHeaderReader.readHeaderFrom(new SeekableFileStream(new
+            File(testVCFFileName)));
+        VariantContextComparator vcfRecordComparator = vcfHeader.getVCFRecordComparator();
+        assertEquals(expectedVariants.size(), actualVariants.size());
+        for (int i = 0; i < expectedVariants.size(); i++) {
+            assertEquals(0, vcfRecordComparator.compare(expectedVariants.get(i),
+                actualVariants.get(i)));
+        }
+    }
+
+    private Path doMapReduce(final Path inputPath) throws Exception {
+        final FileSystem fileSystem = FileSystem.get(conf);
+        final Path outputPath = fileSystem.makeQualified(new Path("target/out"));
+        fileSystem.delete(outputPath, true);
+
+        final Job job = Job.getInstance(conf);
+        FileInputFormat.setInputPaths(job, inputPath);
+
+        job.setInputFormatClass(VCFInputFormat.class);
+        job.setMapOutputKeyClass(LongWritable.class);
+        job.setMapOutputValueClass(VariantContextWritable.class);
+
+        job.setOutputFormatClass(VCFTestWithHeaderOutputFormat.class);
+        job.setOutputKeyClass(LongWritable.class);
+        job.setOutputValueClass(VariantContextWritable.class);
+
+        job.setNumReduceTasks(0);
+        FileOutputFormat.setOutputPath(job, outputPath);
+        if (codecClass != null) {
+            FileOutputFormat.setOutputCompressorClass(job, codecClass);
+        }
+
+        final boolean success = job.waitForCompletion(true);
+        assertTrue(success);
+
+        return outputPath;
+    }
+
+    private static VCFFileReader parseVcf(File vcf) throws IOException {
+        File actualVcf;
+        // work around TribbleIndexedFeatureReader not reading header from .bgz files
+        if (vcf.getName().endsWith(".bgz")) {
```

Comment:
This is only needed for testing. VCFHeaderReader in Hadoop-BAM will correctly read a bgzf stream now (since #97).